VTFLib.NET/VTFFile.cs

```diff
@@ -87,220 +87,228 @@ public struct SVTFCreateOptions
+		[DllImport(dllName, EntryPoint = "vlImageGenerateThumbnail")]
 		public static extern bool ImageGenerateThumbnail(bool srgb);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageGenerateNormalMap")]
+		[DllImport(dllName, EntryPoint = "vlImageGenerateNormalMap")]
 		public static extern bool ImageGenerateNormalMap(uint frame, VTFKernelFilter kernelFilter, VTFHeightConversionMethod heightConversionMethod, VTFNormalAlphaResult normalAlphaResult);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageGenerateAllNormalMaps")]
+		[DllImport(dllName, EntryPoint = "vlImageGenerateAllNormalMaps")]
 		public static extern bool ImageGenerateAllNormalMaps(VTFKernelFilter kernelFilter, VTFHeightConversionMethod heightConversionMethod, VTFNormalAlphaResult normalAlphaResult);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageGenerateSphereMap")]
+		[DllImport(dllName, EntryPoint = "vlImageGenerateSphereMap")]
 		public static extern bool ImageGenerateSphereMap();
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageComputeReflectivity")]
+		[DllImport(dllName, EntryPoint = "vlImageComputeReflectivity")]
 		public static extern bool ImageComputeReflectivity();
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageGetFormatInfo")]
+		[DllImport(dllName, EntryPoint = "vlImageGetFormatInfo")]
 		public static extern ref SVTFImageFormatInfo ImageGetFormatInfo(VTFImageFormat format);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageGetImageFormatInfoEx")]
+		[DllImport(dllName, EntryPoint = "vlImageGetImageFormatInfoEx")]
 		public static extern bool ImageGetImageFormatInfoEx(VTFImageFormat format, ref SVTFImageFormatInfo info);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageComputeImageSize")]
+		[DllImport(dllName, EntryPoint = "vlImageComputeImageSize")]
 		public static extern uint ImageComputeImageSize(uint width, uint height, uint depth, uint mipmapCount, VTFImageFormat format);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageComputeMipmapCount")]
+		[DllImport(dllName, EntryPoint = "vlImageComputeMipmapCount")]
 		public static extern uint ImageComputeMipmapCount(uint width, uint height, uint depth);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageComputeMipmapDimensions")]
+		[DllImport(dllName, EntryPoint = "vlImageComputeMipmapDimensions")]
 		public static extern void ImageComputeMipmapDimensions(uint width, uint height, uint depth, uint mipmapLevel, ref uint mipmapWidth, ref uint mipmapHeight, ref uint mipmapDepth);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageComputeMipmapSize")]
+		[DllImport(dllName, EntryPoint = "vlImageComputeMipmapSize")]
 		public static extern uint ImageComputeMipmapSize(uint width, uint height, uint depth, uint mipmapLevel, VTFImageFormat format);
 
-		[DllImport("VTFLib.dll", EntryPoint = "vlImageConvertToRGBA8888")]
+		[DllImport(dllName, EntryPoint = "vlImageConvertToRGBA8888")]
+		public static extern bool ImageConvertToRGBA8888(IntPtr source, byte[] dest, uint width, uint height, VTFImageFormat sourceFormat);
```

Comment:
made this overload for convenience so I can use the pointer from ImageGetData without having to convert it again .. I might even need to copy data around to get it in the byte array .. suggestions welcome

Reply:
I think you can use unsafe code to convert the IntPtr to a byte array: `byte* src = (byte*)srcPtr.ToPointer()`

Let me know if this works, otherwise you probably have to use `Marshal.Copy`.

Reply:
hmm can't get it to work .. IntPtr to byte* sure, but not to a byte[] (not the same here) and copying is what I was trying to avoid

Reply:
It might be easier to convert byte[] to IntPtr without unsafe code or allocating new memory. 
https://stackoverflow.com/a/23838643

TBH I'm not sure if we should exclusively use IntPtr for these wrapper classes. I will have to do some more research.

Reply:
Is the performance overhead of using `Marshal.Copy` to convert IntPtr to byte[] noticeable? I'd prefer to keep this library consistent if possible.

Reply:
removed it and changed the lumper code to use Marshal.Copy .. didn't notice anything but if there are performance problems in the future we can always revisit this